### PR TITLE
Adding an explainer for real-world hit testing

### PR DESCRIPTION
**Pull request by NellWaliczek**
This PR builds on the great work done by Max, Iker, Blair, and others to clearly define the problem space around adding a real-world hit testing API.  Hopefully the explainer covers all the necessary context, so I'm going to keep this PR description short and sweet.

For clarity's sake, the base for this PR is the input-explainer-cleanup branch.

p.s. I'm still finishing up the PR to add anchor support and will try to get that out asap so folks can better see how the two parts fit together

### CODE REVIEW

File: spatial-tracking-explainer.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ There are three types of reference spaces: `bounded`, `unbounded`, and `stationa
...
 
 ### Spaces, Poses, and transforms
-XR platforms provide a variety of environment-related features such as _hit testing_, _anchors_, _spatial meshes_, _point clouds_, and _markers_. In WebXR, these features, such as `XRReferenceSpace`s, are built on top of `XRSpace`.  An `XRSpace` is an identifiable and trackable point in 3D space that tracking systems use sensor data to continuously update and adjust. On a frame-by-frame basis, developers query the locatability of an `XRSpace` via the `XRFrame.getPose()` function. 
+XR platforms provide a variety of environment-related features such as _hit testing_, _anchors_, _spatial meshes_, _point clouds_, and _markers_. In WebXR, these features, such as `XRReferenceSpace` and [hit testing](hit-teststing-explainer.md), are built on top of `XRSpace`.  An `XRSpace` is an identifiable and trackable point in 3D space that tracking systems use sensor data to continuously update and adjust. On a frame-by-frame basis, developers query the locatability of an `XRSpace` via the `XRFrame.getPose()` function.
```
Comment (toji) [inline spatial-tracking-explainer.md]: Guessing this needs a rebase on top of #22 and #23 still?

Comment (NellWaliczek) [reply spatial-tracking-explainer.md]: Yep.  I'm just waiting on final signoff from you on them.

File: hit-testing-explainer.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+To create an `XRHitTestSource` developers call the `XRSession.requestHitTestSource()` function.  This function accepts an `XRHitTestOptionsInit` dictionary with the following values:
+* The `space` is required and is the `XRSpace` to be tracked by the hit test source.
+* The `offsetRay` is optional and is the `XRRay` from which the hit test should be performed. This ray is defined relative to the `space`. If an `offsetRay` is not provided, the default is an `XRRay` with a forward direction and an origin coincident with the `space` origin.
+
+In this example, an `XRHitTestSource` is created slightly above the center of the `viewerSpace`. This is because the developer has chosen to leave space for buttons along the bottom of the hand-held AR device screen while still giving the perception of a centered cursor (see [Rendering cursors and highlights](#Rendering-cursors-and-highlights)
+
+```js
+let viewerHitTestSource = null;
+
+let hitTestOptions = { space:xrSession.viewerSpace, offsetRay:new XRRay({ origin: {y: 0.5} }) };
```
[the remark this answers is not in the document]
Comment (NellWaliczek) [reply hit-testing-explainer.md]: Ah, my bad.  I'll fix it.

**Comment by NellWaliczek** [comment]
Just got off a call with Iker.  He seems happy with the change in general, though he want to think a bit more on whether the async hit test results should be bound to the frame requested, the latest frame when returned, or both.  I'm comfortable with posting the PR to webxr with that issue still open if you are.

That said... I had the horrible thought that there's an issue i hadn't realized about the relationship between `XRReferenceSpace.originOffset` and the behavior of the hit test apis.  the `originOffset` can be changed at any time during a frame... does that mean that the hit-test results become invalid?  Or do their offsets change?  Should we even allow hit test sources to come from `XRReferenceSpaces`?

**Comment by NellWaliczek** [comment]
Moved to WebXR repo.  Closing the PR, but leaving the branch for now

**Pull request closed without merging**